### ISSUE TITLE
IDE0059 "Remove unnecessary value assignment" is not reported in methods that contain unrelated lambdas

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#49818]
### Analyzer

**Diagnostic ID**: [IDE0059](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0059)

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

IDE0059 bails as soon as a lambda method appears in the same method. This makes the rule flaky. As soon as you start doing even something like `.Select(p => p.Name)`, this analysis is gone for all variables in the method.

### Steps To Reproduce

There should be a IDE0059 diagnostic on the `= 0` assignment, but there is not.

```cs
class C
{
    int M()
    {
        // Comment out this line to see IDE0059 appear on the next line as expected.
        M2(() => { });

        var x = 0;
        x = 1;
        return x;
    }

    void M2(Action action) { }
}
```

**Comment by Youssef1313** [comment]
For some reason this was intentional.

https://github.com/dotnet/roslyn/blob/a6144b2b6accf6c9b0fce4769440b6f0f1bcbb1b/src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs#L29-L33

**Comment by Youssef1313** [comment]
Duplicate of #38376

**Comment by jnm2** [comment]
Thanks @Youssef1313!
